crypto/cipher/aes256cbc/decrypter.go

```diff
@@ -101,3 +105,11 @@ func decryptCBC(key, iv, ciphertext []byte) ([]byte, error) {
 
 	return ret, nil
 }
+
+func validatePrivateKeyType(privateKey crypto.PrivateKey) error {
+	if privateKey.Kind() != crypto.SECP256K1 {
```

Comment:
Checking `.Kind()` can result in someone easily implementing an interface and returning a value, its best to check the `type` is correct. Something like ```
switch pk.(type) {
	case secp256k1.PrivateKey, *secp256k1.PrivateKey:
		return  nil
	default:
		return nil, errors.Errorf("invalid public key type for decrypter")
	}
```
Will ensure the correct type of private key is only allowed.

Reply:
Thanks for the comment. Sure I did the update like that but I only included `*secp256k1.PrivateKey` because since the input argument is an interface, the value has to be pointer of the struct.

Reply:
The methods that return a `PrivateKey` should all return pointers so this is good